Makefile

```diff
@@ -1,58 +1,323 @@
+export ABS_INCLUDEDIR = $(abspath $(INCDIR))
+export ABS_PKGINCLUDEDIR = $(ABS_INCLUDEDIR)/kissfft
+
+#
+# Override LIBDIR with lib64 following CMake's
+# GNUInstallDirs logic:
+#
+
+CANDIDATE_LIBDIR_NAME = lib
+
+ifneq ($(MAKECMDGOALS),clean)
+  ifeq ($(shell uname -s),Linux)
+    _UNAME_ARCH = $(shell uname -i)
+
+    ifeq (,$(_UNAME_ARCH))
+	_UNAME_ARCH = $(shell uname -m)
+
+      ifeq (,$(_UNAME_ARCH))
+	$(warning WARNING: Can not detect system architecture!)
+      endif
+    endif
+
+    ifeq ($(_UNAME_ARCH),x86_64)
+	CANDIDATE_LIBDIR_NAME = lib64
+    endif
+  endif
+endif
+
+CANDIDATE_LIBDIR = $(PREFIX)/$(CANDIDATE_LIBDIR_NAME)
+LIBDIR ?= $(CANDIDATE_LIBDIR)
+
+export ABS_LIBDIR = $(abspath $(LIBDIR))
+
+export INSTALL ?= install
+
+#
+# Library name and version
+#
+
+ifeq ($(KISSFFT_OPENMP), 1)
+	KISSFFTLIB_SHORTNAME = kissfft-$(KISSFFT_DATATYPE)-openmp
+	KISSFFT_PKGCONFIG = kissfft-$(KISSFFT_DATATYPE)-openmp.pc
+	KISSFFTLIB_FLAGS = -fopenmp
+	TYPEFLAGS = -fopenmp
+	PKGCONFIG_OPENMP = -fopenmp
+else
+	KISSFFTLIB_SHORTNAME = kissfft-$(KISSFFT_DATATYPE)
+	KISSFFT_PKGCONFIG = kissfft-$(KISSFFT_DATATYPE).pc
+	TYPEFLAGS =
+	PKGCONFIG_OPENMP =
+endif
+
+ifeq ($(KISSFFT_STATIC), 1)
+	KISSFFTLIB_NAME = lib$(KISSFFTLIB_SHORTNAME).a
+	KISSFFTLIB_FLAGS += -static
+else ifeq ($(shell uname -s),Darwin)
+	KISSFFTLIB_NAME = lib$(KISSFFTLIB_SHORTNAME).dylib
+	KISSFFTLIB_FLAGS += -shared -Wl,-install_name,$(KISSFFTLIB_NAME)
+else
+	KISSFFTLIB_SODEVELNAME = lib$(KISSFFTLIB_SHORTNAME).so
+	KISSFFTLIB_SONAME = $(KISSFFTLIB_SODEVELNAME).$(KFVER_MAJOR)
+	KISSFFTLIB_NAME = $(KISSFFTLIB_SONAME).$(KFVER_MINOR).$(KFVER_PATCH)
+	KISSFFTLIB_FLAGS += -shared -Wl,-soname,$(KISSFFTLIB_SONAME)
+endif
+
+export KISSFFTLIB_SHORTNAME
+
+#
+# Compile-time definitions by datatype
+#
+#
+# Note: -DKISS_FFT_BUILD and -DKISS_FFT_SHARED control
+# C symbol visibility.
+#
+
+ifeq "$(KISSFFT_DATATYPE)" "int32_t"
+	TYPEFLAGS += -DFIXED_POINT=32
+else ifeq "$(KISSFFT_DATATYPE)" "int16_t"
+	TYPEFLAGS += -DFIXED_POINT=16
+else ifeq "$(KISSFFT_DATATYPE)" "simd"
+	TYPEFLAGS += -DUSE_SIMD=1 -msse
+else ifeq "$(KISSFFT_DATATYPE)" "float"
+	TYPEFLAGS += -Dkiss_fft_scalar=$(KISSFFT_DATATYPE)
+else ifeq "$(KISSFFT_DATATYPE)" "double"
+	TYPEFLAGS += -Dkiss_fft_scalar=$(KISSFFT_DATATYPE)
+else
+	$(error ERROR: KISSFFT_DATATYPE must be one of: float double int16_t int32_t simd)
+endif
+
+ifneq ($(KISSFFT_STATIC), 1)
+	TYPEFLAGS += -DKISS_FFT_SHARED
+endif
+
+ifeq ($(KISSFFT_USE_ALLOCA), 1)
+	TYPEFLAGS += -DKISS_FFT_USE_ALLOCA=1
+endif
+
+#
+# Compile-time definitions
+#
+
+#
+# Save pkgconfig variables before appending
+# -DKISS_FFT_BUILD to TYPEFLAGS
+#
+
+ifneq ($(shell uname -s),Darwin)
+	PKGCONFIG_KISSFFT_VERSION = $(KFVER_MAJOR).$(KFVER_MINOR).$(KFVER_PATCH)
+	PKGCONFIG_KISSFFT_OUTPUT_NAME = $(KISSFFTLIB_SHORTNAME)
+	PKGCONFIG_PKG_KISSFFT_DEFS = $(TYPEFLAGS)
+	PKGCONFIG_KISSFFT_PREFIX = $(ABS_PREFIX)
+  ifeq "$(ABS_INCLUDEDIR)" $(abspath "$(PREFIX)/include")
```

Comment:
Hmm, getting the following error now:

Makefile:161: *** invalid syntax in conditional.  Stop.